sdks/java/core/src/main/java/org/apache/beam/sdk/testing/RetryFailures.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.testing;
+
+/** Category tag for tests that apply to runners that retry under failures. */
+public interface RetryFailures {}
```

Comment:
Since your relying on runners to retry failures and support RequiresStableInput, I believe you should add the `UsesTransformThatRequiresStableInput` class as well. This would express the need for a runner to support both to run the tests. Currently your assuming that RetryFailures == Support for RequiresStableInput